### ISSUE TITLE
typescript-urql plugin enforces tsx by default

### ISSUE THREAD

**Issue by xiaoyu-tamu** [dotansimha/graphql-code-generator#4612]
**Describe the bug**
 

User need explicit set `withComponent:false` to use `.ts` extension. but withComponent:false is the default value per doc https://graphql-code-generator.com/docs/generated-config/typescript-urql/#withcomponent.

https://github.com/dotansimha/graphql-code-generator/blob/9d78729e6bb178195bf4f28f8b102144b60d6a5d/packages/plugins/typescript/urql/src/index.ts#L40

**To Reproduce**
```
generates:
  src/graphql.ts:
    config:
      onlyOperationTypes: true
      enumsAsConst: true
      addDocBlocks: false
      preResolveTypes: true
      skipTypename: true
    plugins:
      - typescript
      - typescript-operations
      - typescript-urql
```

---
```
  ✖ src/graphql.ts
    Plugin "urql" requires extension to be ".tsx"!
    Error: Plugin "typescript-urql" validation failed:
```

**Comment by dotansimha** [comment]
Fixed in: https://github.com/dotansimha/graphql-code-generator/pull/4811

**Comment by dotansimha** [comment]
Available in `@graphql-codegen/typescript-urql@2.0.2`
